frontend/src/component/common/ApiTokenTable/useApiTokenTable.tsx

```diff
@@ -61,6 +61,10 @@ export const useApiTokenTable = (
                     <ProjectsList
                         project={props.row.original.project}
                         projects={props.row.original.projects}
+                        isWildcard={props.row.original.secret?.startsWith?.(
```

Comment:
in the backend code we had a concept of orphaned and legacy tokens. Here we use difference vocabulary of legacy and wildcard. What's the reason for that?
Also isX, isY leads to the boolean flags explosion. Can we introduce a union type to represent secret types? e.g. `type SecretType = 'legacy' | 'orphaned'`

Reply:
I agree mostly with this assessment. More than just the number of flags, we also gotta account for what all the combinations mean (legacy && wildcard = ?, legacy && !wildcard = v1, !legacy && wildcard = v2 all projects, !legacy && !wildcard = v2 scoped).

But I'm not sure that `secretType` is the right thing here. Because `orphaned` isn't a secret type. But if we put something like `tokenVersion = v1 | v2` in, then we should be able to take it from there, right? 

if it's v1, it's legacy.
If it's v2, check if it's wildcard or not, etc?

But I'll admit, I'm not quite sure I understand what the payload is for orphaned tokens. @Tymek , can you show me what the payload looks like for:
- legacy tokens
- project scoped v2 tokens
- all project v2 tokens
- orphaned tokens

Reply:
Yup, doesn't have to be SecretType. My main point is that as a reader of the code partially familiar with this part of domain I expect union/list of variants I should be aware of and to have a unit test documenting each of those variants in action.